### ISSUE TITLE
job.server.run_time defaulting to fixed runtime on cmti/cmfe

### ISSUE THREAD

**Issue by raynol-dsouza** [pyiron/pyiron_base#824]
I tried setting a runtime > than the default 345600 (seconds) for a job on the queue, but run_queue.sh still showed 345600/60 (minutes). I took a look at the following setter method in `pyiron_base/pyiron_base/jobs/job/extension/server/generic.py`:

```python
@run_time.setter
def run_time(self, new_run_time):
    """
    The run time in seconds selected for the current simulation
    Args:
        new_run_time (int): run time in seconds
    Raises:
        ValueError: if new_run_time cannot be converted to int
    """
    new_run_time = int(new_run_time)
    if state.queue_adapter is not None and self._active_queue is not None:
        run_time_max = state.queue_adapter.check_queue_parameters(
            queue=self.queue,
            cores=self.cores,
            run_time_max=new_run_time,
            memory_max=self.memory_limit,
        )[1]
        if run_time_max != new_run_time:
            self._run_time = run_time_max
            state.logger.debug("Updated the run time limit to: %i", run_time_max)
        else:
            self._run_time = new_run_time
    else:
        self._run_time = new_run_time
```
and I see that it ignores my new runtime if my queue parameters are specified (which I believe are already specified in `pyiron-resources-mpie/queues/queue.yaml`).

Is this behavior intentional? I would really like to set a runtime larger than the default for some of my VASP jobs...

**Comment by niklassiemer** [comment]
That behavior is intentional. The maximum runtime for a specific queue is configured by the cluster administrator using slurm. For all our queues this is set to (actually already long) four days (=345600s). Therefore, it is not possible to submit jobs to these queues with a longer runtime and this is rejected by this setter. 
Nonetheless, I think the setter could be better!
```python
        if run_time_max != new_run_time:
            self._run_time = run_time_max
            state.logger.debug("Updated the run time limit to: %i", run_time_max)
```
looks like also smaller times than the default (being max-time?) is not used?! Actually, all our calculations run with 4-days?! And I would like to have an actual warning emitted if the user specifies a run time and we change this to another value...

**Comment by raynol-dsouza** [comment]
This definitely makes sense for non-DFT jobs and DFT minimization jobs, but definitely not for DFT-MD runs (which I admit may not (yet) be that common using pyiron). Does one need to obtain permission from the cluster admin to override this runtime to something > than 4 days every time such a longer runtime job needs to be run @freyso?

Yes, by the looks of it there is only a 'fixed' runtime.

**Comment by niklassiemer** [comment]
The fixed runtime is toxic for the scheduler... I performed a lot of AIMD calculations during my PhD and the maximum run_time I had was something in the order of two days. However, there is the possibility of chaining jobs together (not yet in pyiron I fear...)!
My slurm script looked e.g.
```
    #!/bin/bash
    # DO NOT USE environment = COPY_ALL
    #@ job_type = MPICH
    #@ network.MPI = sn_all,not_shared,us
    #@ class = micro
    #@ restart = no
    #@ node = 14
    #@ island_count=1
    #@ tasks_per_node = 4
    #@ wall_clock_limit = 48:00:00
    ##                    00 h 00 min 00 secs
    #@ job_name = llmicro_x2
    #@ initialdir = ./
    #@ output = run.$(jobid)-0.out
    #@ error =  run.$(jobid)-0.err
    #@ energy_policy_tag = h2o_Au_TiO2_node14_omp7_MD
    #@ minimize_time_to_solution = yes
    #@ step_name = R1
    #@ queue

    #@ job_type = MPICH
    #@ network.MPI = sn_all,not_shared,us
    #@ class = micro
    #@ restart = no
    #@ node = 14
    #@ island_count=1
    #@ tasks_per_node = 4
    #@ wall_clock_limit = 48:00:00
    ##                    00 h 00 min 00 secs
    #@ job_name = llmicro_x2
    #@ initialdir = ./
    #@ output = run.$(jobid)-1.out
    #@ error =  run.$(jobid)-1.err
    #@ energy_policy_tag = h2o_Au_TiO2_node14_omp7_MD
    #@ minimize_time_to_solution = yes
    #@ step_name = R2
    #@ dependency = (R1 >= 0)
    #@ queue
```
where the important parts for the chaining are of course 
```
    ...
    #@ step_name = R1
    #@ queue
    
    ...
    #@ step_name = R2
    #@ dependency = (R1 >= 0)
    #@ queue
```

**Comment by samwaseda** [comment]
Yes, this requires a permission every time. Rather than making the run time longer, we should first look into the possibility to re-run calculations. It is only if it is not possible or not reasonable to limit it to 4 days that we should ask the admin to extend it.

**Comment by freyso** [comment]
Yes, presently access to the "longrun" qos is limited to users who asked for it. I think it makes sense to have a separate "queue" setup for such long jobs in pyiron. If I remember correctly, longrun has an upper limit, too (a month? Better check.)

There are several good reason to restrict runtimes for jobs, and in most cases, splitting calculations is possible and useful. In some it isn't, and then longrun is the solution, but this needs to be discussed - longrun-users need to understand my reasons for restricting job runtimes to use this option responsibly, and carefully consider the chance to split jobs.

**Comment by raynol-dsouza** [comment]
I agree with all your points. I wanted to clarify things before I went ahead and did something sketchy. ;D 

@niklassiemer The parameter set that I'm using takes approx. 5 mins per step using 120 cores and my preliminary testing with smaller cells and coarser parameters indicated a run of 4000 steps (2000 thermalization + 2000 actual run). So if I didn't chain jobs (I do not know what this means yet or how it works, I'd definitely be interested in learning), my runtime would easily exceed 4 days.

**Comment by raynol-dsouza** [comment]
@freyso is this the queue you are talking about?

`cmti_large: {cores_max: 1200, cores_min: 40, run_time_max: 345600, script: cmti_large.sh}`

**Comment by raynol-dsouza** [comment]
@samwaseda I'm hoping that there is a way to use the CONTCAR file to continue a VASP job? Couldn't find any documentation on this...

**Comment by samwaseda** [comment]
If `job.restart` doesn't do the job automatically, I don't know. But it's definitely a feature we must have.

**Comment by freyso** [comment]
> @freyso is this the queue you are talking about?
> 
> `cmti_large: {cores_max: 1200, cores_min: 40, run_time_max: 345600, script: cmti_large.sh}`

Yes and no. In principle, cmti_large.sh does specify the qos=longrun option and sets the run-time limit to 29 days. But the queue description in queues.yaml is incorrect (wrong upper limit), and the submission script itself (a) will set two competing time limits if an explicit time limit is given (this is probably a bug, and the 29-day limit likely wins), and (b) will use cmti nodes only - it might be desired to run longrun also on cmfe.
the double-time bug in cmti_large should be fixed, queue description be adapted accordingly. Moreover, a cm_large or cm_long queue should be defined that includes cmfe nodes (within the switch constraint list).

**Comment by niklassiemer** [comment]
I just had a look at the queue adapter and `check_queue_parameters` returns the given run-time except it exceeds the maximum run time of the queue (settings file).

**Comment by niklassiemer** [comment]
What I would change is to 'logger.warn' instead of most of the times silently changing the configuration. And if we set `run_time=None` always the largest possible value is used. I would like to have a better default (12h?) such that only those who need to extend the `run_time` would ask for 4d...

**Comment by freyso** [comment]
Don't change time limits so lightly! It will cause a lot of trouble for people who rely on the current default of 4 days!
In my experience, it is impossible to set up any reasonable default value for run times - it depends so much on the type and size of calculations, and everyone needs to get a feeling for this.
If you apply default values, I suggest to annoy always the user by a warning about the applied time limit, so that people start to set their own (hopefully appropriate) limits (and maybe another warning on collection if the time limit exceeds the actual run time by a factor 10 or more).

**Comment by raynol-dsouza** [comment]
I like both your ideas. :D I suggest the action to be taken be discussed at the CM meeting or pyiron meeting next week.

**Comment by niklassiemer** [comment]
> Don't change time limits so lightly! It will cause a lot of trouble for people who rely on the current default of 4 days!
> In my experience, it is impossible to set up any reasonable default value for run times - it depends so much on the type and size of calculations, and everyone needs to get a feeling for this.

Yes, I absolutely agree! However, pyiron right now just causes people not to think about time limits at all... 

> If you apply default values, I suggest to annoy always the user by a warning about the applied time limit, so that people start to set their own (hopefully appropriate) limits (and maybe another warning on collection if the time limit exceeds the actual run time by a factor 10 or more).

I like the idea to just annoy with warnings 😄
